pkg/util/trace/trace.go

```diff
@@ -0,0 +1,56 @@
+package trace
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/utils/trace"
+)
+
+type Tracer struct {
```

Comment:
What's the reason to create a new library and a new `Tracer` structure instead of creating the trace directly like in the Kubernetes example you pointed out to me?

If I understand correctly, in the Kubernetes code they create a new trace for each function they want to trace, shouldn't we do something similar?

Reply:
The workqueue is multithreaded, so I created a thread safe library for that scenario.  Also, a library will enabled other developers to add tracing to the code pretty easily.